grammar.js

```diff
@@ -790,8 +804,24 @@ module.exports = grammar({
       field('selector', $._type_identifier),
     ),
 
-    function_type: $ => prec.right(seq(
+    match_type: $ => prec.left(PREC.control, seq(
```

Comment:
Is numeric precedence necessary here? I think, we should minimize its usage as it has a very opaque effect on the result of parsing. 
Maybe, relative precedences (opposed to global) would be a better option

Reply:
Good call. At some point this was conflicting with other types, but it seems like I can remove it now.

Reply:
Addressed in 41dcb78.